core/primitives-core/src/profile.rs

```diff
@@ -1,70 +1,36 @@
+use std::rc::Rc;
+use std::{cell::Cell, fmt};
+
 use crate::config::{ActionCosts, ExtCosts};
 use crate::types::Gas;
-use std::{cell::RefCell, fmt, rc::Rc};
-
-const PROFILE_DATA_LEN: usize = 1 + ActionCosts::count() + ExtCosts::count();
-
-/// Data for total gas burnt (index 0), and then each action and external cost
-pub type InternalProfileData<T> = [T; PROFILE_DATA_LEN];
-
-pub struct FixedArray<T> {
-    pub data: Rc<RefCell<InternalProfileData<T>>>,
-}
 
-impl<T> Clone for FixedArray<T> {
-    fn clone(&self) -> Self {
-        Self { data: Rc::clone(&self.data) }
-    }
-}
+/// Data for total gas burnt (index 0), and then each external cost and action
+type DataArray = [Cell<u64>; ProfileData::LEN];
 
 /// Profile of gas consumption.
 #[derive(Clone)]
 pub struct ProfileData {
-    repr: Repr,
```

Comment:
Why this change?

Reply:
This is (was) the main idea of the PR: if we enable profiling by default, we don't need a `Repr`  that can be either enabled or disabled. But, as bowenwang1996 noticed, we probably don't want to execute *anything* not strictly necessary in the runtime, so I'll roll back this particular change, to allow disabling the extra code.